From: Hariprasad Shenai <hariprasad@chelsio.com>
Date: Thu, 17 Dec 2015 13:45:09 +0530
Subject: cxgb4: Use t4_mgmt_tx() API for sending write l2t request ctrl
 packets.
Patch-mainline: v4.5-rc1
Git-commit: 9baeb9d7d8f61f7786416e4f001d738750d2aa24
References: bsc#963896 FATE#320113

Based on original work by Michael Werner <werner@chelsio.com>

Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/chelsio/cxgb4/l2t.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/net/ethernet/chelsio/cxgb4/l2t.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/l2t.c
@@ -161,8 +161,7 @@ static int write_l2e(struct adapter *ada
 		memcpy(e->dmac, e->neigh->ha, sizeof(e->dmac));
 	memcpy(req->dst_mac, e->dmac, sizeof(req->dst_mac));
 
-	set_wr_txq(skb, CPL_PRIORITY_CONTROL, 0);
-	t4_ofld_send(adap, skb);
+	t4_mgmt_tx(adap, skb);
 
 	if (sync && e->state != L2T_STATE_SWITCHING)
 		e->state = L2T_STATE_SYNC_WRITE;
